frontend/src/components/SessionTimer.js

```diff
@@ -0,0 +1,224 @@
+import styled from 'styled-components/macro';
+import { useState } from 'react';
+import ButtonBig from './ButtonBig';
+import ButtonTab from './ButtonTab';
+import { GiMeeple } from 'react-icons/all';
+
+export default function SessionTimer({
+    setStartDate,
+    setSessionDuration,
+    session,
+}) {
+    const [timer, setTimer] = useState({ s: 0, m: 0, h: 0 });
+    const [startTimeForSession, setStartTimeForSession] = useState('');
+    const [stopTimeForSession, setStopTimeForSession] = useState('');
+    const [intervalId, setIntervalId] = useState();
+    const [gameHasStarted, setGameHasStarted] = useState(false);
+    const [duration, setDuration] = useState();
+
+    const onClickToStart = () => {
+        let startDate = new Date();
+        let startDateTimeStamp = startDate.getTime();
+        setStartTimeForSession(startDateTimeStamp);
+        setStartDate(startDate);
+        console.log(startDate);
+        console.log(startDateTimeStamp);
+        run();
+        const intervalId = setInterval(run, 1000);
+        setIntervalId(intervalId);
+        setGameHasStarted(true);
+    };
+
+    const onClickToStop = () => {
+        let stopDate = new Date();
+        let stopDateTimeStamp = stopDate.getTime();
+        console.log(stopDate);
+        setStopTimeForSession(stopDateTimeStamp);
+        console.log(stopTimeForSession);
+
+        clearInterval(intervalId);
+        setGameHasStarted(false);
+        const gameDuration = (
+            (new Date().getTime() - startTimeForSession) /
+            1000 /
+            60
+        )
+            .toString()
+            .split('.')[0];
+
+        console.log(gameDuration);
+        setDuration(gameDuration);
+    };
+
+    let updatedS = timer.s,
+        updatedM = timer.m,
+        updatedH = timer.h;
```

Comment:
It's way easier to just keep one state with seconds and derive minutes and hours from that!

Reply:
oh, right... makes sense. i might even try to rely on timestamps instead of of doing seconds++